qmcpy/integrand/_integrand.py

```diff
@@ -3,21 +3,26 @@
 from ..discrete_distribution._discrete_distribution import DiscreteDistribution
 from numpy import *
 import os
-import multiprocessing
+from multiprocessing import get_context
+from multiprocessing.pool import ThreadPool
 from itertools import repeat
 
 
 class Integrand(object):
     """ Integrand abstract class. DO NOT INSTANTIATE. """
 
-    def __init__(self, dimension_indv, dimension_comb, parallel):
+    def __init__(self, dimension_indv, dimension_comb, parallel, threadpool=False):
         """
         Args:
             dimension_indv (tuple): individual solution shape.
             dimension_comb (tuple): combined solution shape. 
             parallel (int): If parallel is False, 0, or 1: function evaluation is done in serial fashion.
-                Otherwise, parallel specifies the number of CPUs used by multiprocessing.Pool.
-                Passing parallel=True sets the number of CPUs equal to os.cpu_count().
+                Otherwise, parallel specifies the number of processes used by 
+                multiprocessing.Pool or multiprocessing.pool.ThreadPool.
+                Passing parallel=True sets processes = os.cpu_count().
+            threadpool (bool): When parallel > 1, 
+                if use_threads = True then use multiprocessing.pool.ThreadPool 
```

Comment:
I think that's supposed to be "if threadpool = True"?

Reply:
Yes, thanks for catching this.